10.0/Dockerfile

```diff
@@ -93,9 +80,15 @@ RUN { \
 	&& apt-get update \
 	&& apt-get install -y \
 		"mariadb-server=$MARIADB_VERSION" \
-# percona-xtrabackup is installed at the same time so that `mysql-common` is only installed once from just mariadb repos
-		percona-xtrabackup \
-		socat \
+		 \
+	# Ensure xtrabkacup SST continue to work, so we don't break existing setups.
+	&& (test ! -f /usr/bin/mariabackup && true \
```

Comment:
Doesn't this imply that anywhere mariadb-backup isn't supported (like here in 10.0) that xtrabackup is also no longer installed and thus users will be broken?  I don't think this is acceptable as-is, sorry.

It would also be great if we could get someone from MariaDB upstream to weigh in on whether this hacking of `mariabackup` to pretend it's `xtrabackup` is something they're willing to support (I like the idea conceptually, but it appears that it is not a 100% drop-in replacement from what's implemented here). :sweat_smile:

Reply:
@vaintroub want to add an official upstream comment here?

Reply:
I cannot officially speak about what we're willing to support and what not, product management is for that . To give the perspective of mariabackup - it is basically parameter compatible with xtrabackup 2.3, just the home-made encryption is not supported, nor the tar streaming format. Unknown parameters are silently ignored, mariabackup inherited this ingenious feature  from  xtrabackup. There are no plans to make mariabackup command-line incompatible, and we'll try to fix the command-line incompatibilities. There should be no dependency on binary name either, so hacking to pretend should work fine, in most cases

Reply:
> Doesn't this imply that anywhere mariadb-backup isn't supported (like here in 10.0) that xtrabackup is also no longer installed and thus users will be broken? I don't think this is acceptable as-is, sorry.

`xtrabackup` was added for Galera (10.1+) SST support and all `xtrabackup` -> `mariabackup` compatibility logic was done, so we don't break existing Galera setups which use `xtrabackup` for SST. So if decide to remove `xtrabackup` from <10.1, it won't break existing MySQL setups but only setup where `xtrabackup` is used for backups.

Reply:
Additionally it looks like the plan is to make mariabackup-10.1 work with 10.0 and 5.5 (MDEV-14937, though MDEV-14176 is an outstanding issue for 5.5). When that happens can add mariabackup back into <10.1 if serving the backup functionality is required.